code-challenges.test.js

```diff
@@ -13,34 +16,123 @@
 
 const people = [
   { name: "ford prefect", occupation: "a hitchhiker" },
   { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
   { name: "arthur dent", occupation: "a radio employee" }
 ]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+
+// Create describe statement
+describe("hitchhiker", () => {
+  // It statement includes what we want to occur
+  it("returns an array with a sentence about each person with their name capitalized", () => {
+    // Our expected output based on the It statement
+    expect(hitchhiker(people)).toEqual([
+      "Ford Prefect is a hitchhiker.",
+      "Zaphod Beeblebrox is president of the galaxy.",
+      "Arthur Dent is a radio employee.",
+    ])
+  })
+})
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 total
+// Snapshots:   0 total
+// Time:        0.169 s
 
 // b) Create the function that makes the test pass.
 
+// Create function that passes an array
+const hitchhiker = (array) => {
+  // Iterate over the array using .map()
+  return array.map((value) => {
+    // We'll iterate over each element by splitting the words and returning each first letter as capitalized, then concatenating the remainder of the word
+    const towelPeeps = value.name.split(" ").map((value) => {
+        return value[0].toUpperCase() + value.substring(1)}).join(" ")
+        // We can return our towelPeeps variable and their respective occupations with string interpolation
+    return `${towelPeeps} is ${value.occupation}.`
+  })
+}
+
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+// Input: Variables provided
 // Expected output: [ 2, 0, -1, 0 ]
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
 // Expected output: [ 2, 1, -1 ]
 
+const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+
+
+// Create describe statement
+describe("remainders", () => {
+  // It statement includes what we want to occur
+  it("takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3", () => {
+    // Our expected output based on the It statement
+    expect(remainders(hodgepodge1)).toEqual([2, 0, -1, 0])
+    expect(remainders(hodgepodge2)).toEqual([2, 1, -1])
+  })
+})
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 total
+// Snapshots:   0 total
+// Time:        0.219 s, estimated 1 s
+
 // b) Create the function that makes the test pass.
 
+// Create function that takes in an array
+const remainders = (array) => {
+  // We'll use .filter() to take out any string in the array and give us only numbers
```

Comment:
Nice, I like this approach and the pseudocode in between that tells us what's going on! 👏🏽